src/api/pool.go

```diff
@@ -0,0 +1,109 @@
+package api
+
+import (
+	"context"
+	"github.com/go-resty/resty/v2"
+	"golang.org/x/sync/errgroup"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
+	client          *Client         // http client
+	ctx             context.Context // context of the parallel work
+	workers         *errgroup.Group // error group -> if one worker fails, all will be stopped
+	counter         sync.WaitGroup  // detect when all requests are processed
```

Comment:
Najtazsie je urcit, kedy sme spracovali vsetky requesty a mozeme skoncit, ... kedze pri spracovani `response` sa mozu generovat dalsie `requests`. Preto je tu tato `sync.WaitGroup` .... ma metody `Add` a `Done` ... `Add` pripocita `1` a `Done` odpocita `1`.
Ak je stav pocitadla `0`, tak to znamena, ze sme vsetko spracovali, pretoze pri novom `requeste` sa vola `Add` a pri dokonceni spracovania `response` (alebo chybe) sa vola `Done`.

`sync.WaitGroup` ma metodu `Wait`, ktora caka, pokial na pocitadle nebude nula a az potom pokracuje beh programu.